packages/splits-sdk/src/client/data.ts

```diff
@@ -0,0 +1,1061 @@
+          relatedAccounts.upstreamLiquidSplits.map((gqlLiquidSplit) =>
+            formatGqlSplit(gqlLiquidSplit.split),
+          ),
+        )
+      }
+
+      response.controllingSplits =
+        relatedAccounts.controllingSplits?.map((split) =>
+          formatGqlSplit(split),
+        ) ?? []
+      response.pendingControlSplits =
+        relatedAccounts.pendingControlSplits?.map((split) =>
+          formatGqlSplit(split),
+        ) ?? []
+      response.ownedSwappers =
+        relatedAccounts.ownedSwappers?.map((swapper) =>
+          formatGqlSwapper(swapper),
+        ) ?? []
+      response.ownedPassThroughWallets =
+        relatedAccounts.ownedPassThroughWallets?.map((passThroughWallet) =>
+          formatGqlPassThroughWallet(passThroughWallet),
+        ) ?? []
+
+      response.account = formatFullGqlAccount(
+        result.account,
+        response.upstreamSplits,
+        response.upstreamVesting,
+        response.upstreamWaterfalls,
+        response.upstreamLiquidSplits,
+        response.upstreamSwappers,
+        response.upstreamPassThroughWallets,
+        relatedAccounts.controllingSplits,
+        relatedAccounts.pendingControlSplits,
+        relatedAccounts.ownedSwappers,
+        relatedAccounts.ownedPassThroughWallets,
+      )
+
+      const allPassThroughWallets = [...relatedAccounts.ownedPassThroughWallets]
+      if (result.account.__typename === 'PassThroughWallet') {
+        allPassThroughWallets.push(result.account)
+      }
+
+      if (result.account.__typename === 'LiquidSplit') {
+        // Not really an upstream split, but it's just used for loading. Should probably update that name. Maybe related splits?
+        response.upstreamSplits.push(formatGqlSplit(result.account.split))
+      } else if (result.account.__typename === 'WaterfallModule') {
+        result.account.tranches.map((gqlWaterfallTranche) => {
+          if (gqlWaterfallTranche.recipient.__typename === 'Split') {
+            response.upstreamSplits = response.upstreamSplits ?? []
+            response.upstreamSplits.push(
+              formatGqlSplit(gqlWaterfallTranche.recipient),
+            )
+          }
+        })
+      }
+
+      if (allPassThroughWallets.length > 0) {
+        allPassThroughWallets.map((gqlPassThroughWallet) => {
+          if (gqlPassThroughWallet.passThroughAccount.__typename === 'Split') {
+            response.upstreamSplits = response.upstreamSplits ?? []
+            response.upstreamSplits.push(
+              formatGqlSplit(gqlPassThroughWallet.passThroughAccount),
+            )
+            gqlPassThroughWallet.passThroughAccount.recipients.map(
+              (gqlRecipient) => {
+                if (gqlRecipient.account.__typename === 'Swapper') {
+                  response.upstreamSwappers = response.upstreamSwappers ?? []
+                  response.upstreamSwappers.push(
+                    formatGqlSwapper(gqlRecipient.account),
+                  )
```

Comment:
yeah, I actually didn't want to change any of this because I think as we discussed we want to move the subgraph stuff from the client to the sdk so it can be reused elsewhere, so tried to copy things as is for now, happy to shed stuff out if needed though, lmk what you think